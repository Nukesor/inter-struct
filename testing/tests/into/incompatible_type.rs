use inter_struct::InterStruct;

/// This should crash, since the types for the IntoStruct are incompatible.
#[derive(InterStruct)]
#[into("crate::IntoStruct")]
pub struct FromStruct {
    pub normal: i32,
    pub optional: Option<i32>,
}

pub struct IntoStruct {
    pub normal: String,
    pub optional: Option<String>,
}

fn main() {}
