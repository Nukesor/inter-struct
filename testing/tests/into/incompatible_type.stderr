error: Type 'i32 cannot be merged into field of type 'String'.
 --> tests/into/incompatible_type.rs:7:17
  |
7 |     pub normal: i32,
  |                 ^^^

error: Type 'Option < i32 > cannot be merged into field of type 'String'.
 --> tests/into/incompatible_type.rs:8:19
  |
8 |     pub optional: Option<i32>,
  |                   ^^^^^^

error[E0063]: missing fields `normal` and `optional` in initializer of `IntoStruct`
 --> tests/into/incompatible_type.rs:5:8
  |
5 | #[into("crate::IntoStruct")]
  |        ^^^^^^^^^^^^^^^^^^^ missing `normal` and `optional`
